graph/telemetry/istio/appender/ambient.go

```diff
@@ -62,36 +62,42 @@ func (a AmbientAppender) handleWaypoints(trafficMap graph.TrafficMap, globalInfo
 			waypointName = n.Service
 		}
 		if isWaypoint(&waypoints, n.Cluster, n.Namespace, waypointName) {
+			waypointsNodes[n.ID] = true
 			if !a.ShowWaypoints {
-				deletedWaypoints[n.ID] = true
 				delete(trafficMap, n.ID)
 			} else {
 				n.Metadata[graph.IsOutOfMesh] = false
 				n.Metadata[graph.IsWaypoint] = true
-				n.Edges = []*graph.Edge{}
+				for _, edge := range n.Edges {
+					// Just hide so we have all the information
+					edge.Metadata[graph.Display] = "hide"
+				}
 			}
 		}
 	}
 
-	if len(deletedWaypoints) > 0 {
+	if len(waypointsNodes) > 0 {
 		for _, n := range trafficMap {
-			n.Edges = sliceutil.Filter(n.Edges, func(edge *graph.Edge) bool {
-				return !deletedWaypoints[edge.Dest.ID]
-			})
+			// Delete edges
+			if !a.ShowWaypoints {
+				n.Edges = sliceutil.Filter(n.Edges, func(edge *graph.Edge) bool {
+					return !waypointsNodes[edge.Dest.ID]
+				})
+			}
+
+			// Find duplicates
+			for _, edge := range n.Edges {
+				// If not show the waypoint, mark the nodes
+				if a.ShowWaypoints {
+					if waypointsNodes[edge.Dest.ID] {
+						edge.Metadata[graph.Display] = "reverse"
+					}
+				}
+			}
```

Comment:
```suggestion
			if !a.ShowWaypoints {
				n.Edges = sliceutil.Filter(n.Edges, func(edge *graph.Edge) bool {
					return !waypointsNodes[edge.Dest.ID]
				})
				continue
			}

			// Find duplicates
			for _, edge := range n.Edges {
				if waypointsNodes[edge.Dest.ID] {
					edge.Metadata[graph.Display] = "reverse"
				}
			}
```

I think this can be simplified, we already know `a.showWaypoints` is true.